submissions/bmukha/frogger/app.js

```diff
@@ -1,149 +1,192 @@
+    }
+  }
+}
+
+class Enemy {
+  constructor(startY, player, env, scoreBoard) {
+    this.scoreBoard = scoreBoard;
+    this.env = env;
+    this.directions = ['left', 'right'];
+    this.direction = this.env.getRandomDirection();
+    this.sprite = `images/enemy-bug-${this.direction}.png`;
+    this.minX = -this.env.tileWidth;
+    this.maxX = this.env.fieldMaxX + this.env.tileWidth;
+    this.startX = this.direction === 'right' ? this.minX : this.maxX;
+    this.startY = startY;
+    this.currX = this.startX;
+    this.currY = this.startY;
+    this.minSpeed = 150;
+    this.maxSpeed = 300;
+    this.mult =
+      this.direction === 'right'
+        ? this.env.getRandomNumber(this.minSpeed, this.maxSpeed)
+        : -this.env.getRandomNumber(this.minSpeed, this.maxSpeed);
+    this.player = player;
+  }
+  checkCollision() {
+    if (this.player.currY !== this.currY) {
+      return false;
+    }
+    if (
+      (this.player.currX >= this.currX && this.player.currX <= this.currX + this.env.tileWidth) ||
+      (this.player.currX + this.env.tileWidth >= this.currX &&
+        this.player.currX + this.env.tileWidth <= this.currX + this.env.tileWidth)
+    ) {
+      return true;
+    }
+    return false;
+  }
+  update(dt) {
+    if (this.checkCollision()) {
+      this.scoreBoard.resetScore();
+      this.scoreBoard.updateGameInfo(this.scoreBoard.loseMessage);
+      this.player.update(this.player.startX, this.player.startY);
       setTimeout(() => {
-        player.update(this.startX, this.startY);
-        infoHolder.innerText = `Let's go!`;
-      }, 1000);
-      if (currentScore > highScore) {
-        highScore = currentScore;
-        localStorage.setItem('highScore', highScore);
-        highHolder.innerText = `High score: ${highScore}`;
-      }
+        this.scoreBoard.updateGameInfo(this.scoreBoard.welcomeMessage);
+      }, 500);
     }
+    if (this.currX >= this.maxX) {
+      this.mult *= -1;
+      this.direction = 'left';
+      this.sprite = `images/enemy-bug-${this.direction}.png`;
+    } else if (this.currX <= this.minX) {
+      this.mult *= -1;
+      this.direction = 'right';
+      this.sprite = `images/enemy-bug-${this.direction}.png`;
+    }
+    this.currX = this.currX + dt * this.mult;
+  }
+  render() {
+    ctx.drawImage(Resources.get(this.sprite), this.currX, this.currY);
   }
 }
 
+const env = new Environment();
+const scoreBoard = new ScoreBoard('.high', '.message', '.current');
+const player = new Player(env, scoreBoard);
+const firstEnemyY = env.fieldMinY + env.tileHeight;
+const secondEnemyY = env.fieldMinY + env.tileHeight * 2;
+const thirdEnemyY = env.fieldMinY + env.tileHeight * 3;
 const allEnemies = [
-  new Enemy(65, getRandomNumber(150, 300)),
-  new Enemy(150, getRandomNumber(150, 300)),
-  new Enemy(235, getRandomNumber(150, 300)),
+  new Enemy(firstEnemyY, player, env, scoreBoard),
+  new Enemy(secondEnemyY, player, env, scoreBoard),
+  new Enemy(thirdEnemyY, player, env, scoreBoard),
```

Comment:
What if I need to add 20 more enemies?
Do I need to define `fourthEnemyY` etc or this `Y`'s are actually y-coords for the field rows?